src/lib.rs

```diff
@@ -53,112 +54,137 @@ pub const MAX_LENGTH: usize = sys::MAX_LENGTH;
 /// [`MAX_LENGTH`] to an `IoVec` will result in a panic.
 ///
 /// [`MAX_LENGTH`]: constant.MAX_LENGTH.html
-pub struct IoVec {
+pub struct IoVec<'a> {
     sys: sys::IoVec,
+    _p: PhantomData<&'a [u8]>,
 }
 
-impl IoVec {
-    pub fn from_bytes(slice: &[u8]) -> Option<&IoVec> {
-        if slice.len() == 0 {
-            return None
+/// Mutable byte slice type for performing vectored I/O operations.
+pub struct IoVecMut<'a> {
+    sys: sys::IoVec,
+    _p: PhantomData<&'a mut [u8]>,
+}
+
+// ===== impl IoVec =====
+
+impl<'a> IoVec<'a> {
+    /// Convert an `IoVec` from a byte slice
+    pub fn from_bytes(slice: &[u8]) -> Self {
```

Comment:
This will probably want to use an explicit lifetime (`'a`) as right now I believe this is not memory safe.

Reply:
I think you are correct.